File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    const newNums = [];
    newNums.push(nums[0]);
    
    for(let i = 1; i < nums.length; i++){
        newNums.push(nums[i]+newNums[i-1]);
    }
    return newNums;    
};

File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {
        
        int prefSum = nums[0];
        int[] res = new int[nums.length];
        res[0] = prefSum;
        
        for(int i = 1; i<nums.length; i++){
            prefSum+=nums[i];
            res[i] = prefSum;
        }
        
        return res;
        
    }
}